src/main/kotlin/me/lensvol/blackconnect/BlackdClient.kt

```diff
@@ -14,12 +14,15 @@ sealed class BlackdResponse {
     data class UnknownStatus(val code: Int, val responseText: String) : BlackdResponse()
 }
 
-class BlackdClient(val hostname: String, val port: Int) {
+class BlackdClient(val hostname: String, val port: Int, val https: Boolean) {
 
     private val logger = Logger.getInstance(BlackdClient::class.java.name)
 
     fun checkConnection(): Result<String, String> {
-        val url = URL("https://$hostname:$port")
+        var url = URL("http://$hostname:$port")
+        if (https) {
```

Comment:
val protocol = if (useSsl) "https" else "http
val url = URL("$protocol://$hostname:$port")

Reply:
thx this Java Kotlin code is not my best friend :D didnt want to dive too deep. I'll adjust